lib/ui/painting/flutter_rtree.h

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_FLUTTER_RTREE_H_
+#define FLUTTER_LIB_UI_PAINTING_FLUTTER_RTREE_H_
+
+#include "third_party/skia/include/core/SkBBHFactory.h"
+#include "third_party/skia/include/core/SkTypes.h"
+
+namespace flutter {
+/**
+ * An R-Tree implementation based on
+ * https://chromium.googlesource.com/skia/+/master/src/core/SkRTree.h. In short,
+ * it is a balanced n-ary tree containing a hierarchy of bounding rectangles.
+ *
+ * It only supports bulk-loading, i.e. creation from a batch of bounding
+ * rectangles. This performs a bottom-up bulk load using the STR
+ * (sort-tile-recursive) algorithm.
+ *
+ * TODO: Experiment with other bulk-load algorithms (in particular the Hilbert
+ * pack variant, which groups rects by position on the Hilbert curve, is
+ * probably worth a look). There also exist top-down bulk load variants
+ * (VAMSplit, TopDownGreedy, etc).
+ *
+ * For more details see:
+ *
+ *  Beckmann, N.; Kriegel, H. P.; Schneider, R.; Seeger, B. (1990). "The
+ * R*-tree: an efficient and robust access method for points and rectangles"
+ */
+class FlutterRTree : public SkBBoxHierarchy {
```

Comment:
The Flutter prefix in the `FlutterRTree` is somewhat redundant. Please rename this to something more descriptive of the utility of this object.